### PR TITLE
feat: add experimental support for TTL

### PR DESCRIPTION
**Pull request by ascorbic**
### Summary

Adds experimental support for ttl in pages with revalidate set. Parses the cache-control header returned by Next and extracts the `s-maxage` value, which it then uses in the ttl response. This isnlt enabled by default, but requires an env var flag.

As part of the build process we need to patch the `next-server.js` file, because we need to disable caching for ISR pages and currently this is only supported for `minimalMode`. We do this by adding another env var to the line where it checks for all the conditions for disabling the SSG cache.

### Test plan

1. Visit [a page with revalidate set](https://deploy-preview-833--netlify-plugin-nextjs-demo.netlify.app/getStaticProps/withRevalidate/1/). The rendered-at time will probably be old. If the rendered-at time is the current wall time then it's a fresh deploy so wait a minute for it to be stale then load again.
2. Wait for 1 minute for the revalidated content to propagate to your edge node
3. Load the page again. The rendered-at time should match the time at step 1 (so about a minute in the past)

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

image

### CODE REVIEW

File: src/templates/handlerUtils.ts [hunk 1]
```diff
@@ -31,3 +31,33 @@ export const downloadFile = async (url, destination) => {
...
+  return 0
+}
+
+export const getMultiValueHeaders = (
+  headers: Record<string, string | Array<string>>,
+): Record<string, Array<string>> => {
+  const multiValueHeaders: Record<string, Array<string>> = {}
+  for (const key of Object.keys(headers)) {
+    if (Array.isArray(headers[key])) {
+      multiValueHeaders[key] = headers[key] as Array<string>
```
Comment (tiffanosaurus) [inline src/templates/handlerUtils.ts:57]: You could do the following instead so you don't need to coerce

```ts
    const header = headers[key];

    if (Array.isArray(header)) {
      multiValueHeaders[key] = header
    } else {
      multiValueHeaders[key] = [header]
    }
```

File: src/helpers/files.js [hunk 2]
```diff
@@ -219,6 +219,57 @@ exports.moveStaticPages = async ({ netlifyConfig, target, i18n }) => {
   }
 }
 
+const patchFile = async ({ file, from, to }) => {
```
Comment (tiffanosaurus) [inline src/helpers/files.js:222]: ooo neat

**Review by tiffanosaurus — APPROVED** [review]
Sorry for the wait, I thought I clicked the thing. Looks awesome 😮!

File: src/helpers/files.js (hunk 2 above)
Comment (ascorbic) [reply src/helpers/files.js:222]: I hate it!

**Pull request merged**